internal/js_library/js_library.bzl

```diff
@@ -223,10 +223,12 @@ def _impl(ctx):
             if DefaultInfo in dep:
                 files_depsets.append(dep[DefaultInfo].files)
 
+    # We do not include typings_depsets in the runfiles because that would cause type-check actions to occur
+    # in every development workflow.
     runfiles = ctx.runfiles(
         files = all_files,
         transitive_files = depset(
-            transitive = files_depsets + typings_depsets,
+            transitive = files_depsets,
```

Comment:
this reverts https://github.com/bazelbuild/rules_nodejs/pull/2547
which I think was a mistake, since it makes type-checking eager for a js_library, and it shouldn't be.
/cc @Aghassi

Reply:
Reverted in https://github.com/bazelbuild/rules_nodejs/pull/3189 for reference